postgres/datadog_checks/postgres/metadata.py

```diff
@@ -388,20 +388,30 @@ def _get_table_info(self, cursor, dbname, schema_id):
 
         If any tables are partitioned, only the master paritition table name will be returned, and none of its children.
         """
+        if VersionUtils.transform_version(str(self._check.version))["version.major"] == "9":
+            cursor.execute(PG_TABLES_QUERY_V9.format(schema_oid=schema_id))
+        else:
+            cursor.execute(PG_TABLES_QUERY_V10_PLUS.format(schema_oid=schema_id))
+        rows = cursor.fetchall()
+        table_info = [dict(row) for row in rows]
+
         limit = self._config.schemas_metadata_config.get("max_tables", 300)
-        if self._config.relations:
-            if VersionUtils.transform_version(str(self._check.version))["version.major"] == "9":
-                cursor.execute(PG_TABLES_QUERY_V9.format(schema_oid=schema_id))
-            else:
-                cursor.execute(PG_TABLES_QUERY_V10_PLUS.format(schema_oid=schema_id))
-            rows = cursor.fetchall()
-            table_info = [dict(row) for row in rows]
-            return self._sort_and_limit_table_info(cursor, dbname, table_info, limit)
 
-        else:
-            # Config error should catch the case where schema collection is enabled
-            # and relation metrics aren't, but adding a warning here just in case
-            self._check.log.warning("Relation metrics are not configured for {dbname}, so tables cannot be collected")
+        if len(table_info) <= limit:
+            return table_info
+        if not self._config.relations:
+            self._check.log.warning(
+                "Number of tables exceeds limit of {limit} set by max_tables but "
+                "relation metrics are not configured for {dbname}."
+                "Please configure relations to collect metrics for all tables."
```

Comment:
since this error would be about schema collection, should it say something like "some schemas will be missing, please configure relations to collect only the most active tables"? i think enabling relation metrics would still have a missing schemas problem if beyond the max gets collected